File: surrealml/clients/python/surrealml/engine/onnx.py
"""
This file defines the adapter for the ONNX file format. This adapter does not convert anything as the input
model is already in the ONNX format. It simply saves the model to a file. However, I have added this adapter
to keep the same structure as the other adapters for different engines (maxwell flitton).
"""
from surrealml.engine.utils import create_file_cache_path


class OnnxAdapter:

    @staticmethod
    def save_model_to_onnx(model, inputs) -> str:
        """
        Saves a model to an onnx file.

        :param model: the raw ONNX model to directly save
        :param inputs: the inputs to the model needed to trace the model
        :return: the path to the cache created with a unique id to prevent collisions.
        """
        file_path = create_file_cache_path()

        with open(file_path, "wb") as f:
            f.write(model.SerializeToString())

        return file_path

File: surrealml/clients/typescript/src/surrealml/engine/onnx.ts
/**
 * This adapter handles the ONNX file format.
 * The input model is already in ONNX format; this simply saves it to disk.
 * Kept for structural consistency with other engine adapters.
 */
import fs from 'fs';
import { createFileCachePath } from './utils.js';


export class OnnxAdapter {
    /**
     * Saves a model to an ONNX file.
     *
     * model: raw onnx protobuf bytes
     * returns: the path to the cache file (unique to prevent collisions)
     */
	static saveModelToOnnx(model: Uint8Array): string {
		const filePath = createFileCachePath();      
		fs.writeFileSync(filePath, model);  
		return filePath;
	  }	
}
